java-squid/src/main/java/org/sonar/java/model/expression/MethodInvocationTreeImpl.java

```diff
@@ -44,12 +44,17 @@
   @Nullable
   private TypeArguments typeArguments;
   private Symbol symbol = Symbols.unknownSymbol;
+  private SyntaxToken openParenToken;
+  private SyntaxToken closeParenToken;
 
-  public MethodInvocationTreeImpl(ExpressionTree methodSelect, @Nullable TypeArguments typeArguments, List<ExpressionTree> arguments, AstNode... children) {
+  public MethodInvocationTreeImpl(ExpressionTree methodSelect, SyntaxToken openParenToken, SyntaxToken closeParenToken, @Nullable TypeArguments typeArguments,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)org/sonar/java/model/expression/MethodInvocationTreeImpl must not use com/sonar/sslr/api/AstNode[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)